nevergrad/optimization/base.py

```diff
@@ -314,6 +314,8 @@ def tell(self, candidate: p.Parameter, loss: tp.Loss) -> None:
         if isinstance(loss, (Real, float)):
             # using "float" along "Real" because mypy does not understand "Real" for now Issue #3186
             loss = float(loss)
+            if not (loss < 5.0e20):
+                loss = 5.0e20  # sys.float_info.max leads to numerical problems so let us do this.
```

Comment:
What happens beyond?
I really don't like that, this is a hard constant and can lead to unexpected behaviors. It does make a hard assumption on the cost function.
If there is no way out of it, at the very least:
1. add an explanation as a comment
2. use warnings.warn to send a warning that there is something not going well, this must not go unnoticed.

Reply:
Warning added.
Explanation added.